style(replays): altered chart style to use out of the box styles and adjusted y-axis to be rounded values

Danondso opened the pull request:
# Description
Closes #34110 

# Screenshot
<img width="891" alt="image" src="https://user-images.githubusercontent.com/7014871/166335591-51154b01-2f80-41b6-9e86-c5481a0184fa.png">

<img width="1018" alt="image" src="https://user-images.githubusercontent.com/7014871/166335821-e2fcafcc-5631-4d99-a0fb-519d364d4f21.png">



<!-- Describe your PR here. -->



<!--

  The following applies to third-party contributors.
  Sentry employees and contractors can delete or ignore.

-->

----

By submitting this pull request, I confirm that Sentry can use, modify, copy, and redistribute this contribution, under Sentry's choice of terms.

Danondso committed:
style: altered chart style to use out of the box styles
refactor: rounded the values shown on the y-axis to the nearest tenth.

Danondso committed:
Merge branch 'master' into style/replays/34110/heap-memory-chart-followup

ryan953 commented on static/app/views/replays/detail/memoryChart.tsx:
```diff
@@ -16,30 +20,37 @@ type Props = {
-    },
-    xAxis: {
+    }),
+    xAxis: XAxis({
       type: 'category',
       min: formatTimestamp(memorySpans[0]?.timestamp),
```
The docs show that `'time'` is a valid value: https://echarts.apache.org/en/option.html#xAxis.type
idk if this is the same version that we have installed though

```suggestion
    xAxis: XAxis({
      type: 'time',
      min: formatTimestamp(memorySpans[0]?.timestamp),
```

Danondso replied:
I didn't have any trouble when all the values were numbers but when I format the timestamps the chart doesn't want to render the data if it set it to 'time'. I could try to dig into more if you'd like.

ryan953 replied:
Could it be that the `min` and `max` should be numbers when type=time?

Sorry, i hadn't noticed that `'time'` was the thing that made break last time. Probably not worth spending all the time on it now.

Danondso replied:
All good, I tried time without formatting the min and max and it still didn't show. 

```
    xAxis: XAxis({
      type: 'time',
      min: memorySpans[0]?.timestamp,
      max: memorySpans[memorySpans.length - 1]?.timestamp,
      axisLabel: {
        formatter: value => formatTimestamp(value),
      },
      theme,
    }),
```

Danondso replied:
Though if I didn't format the name the x-axis ticks did show up. I'm not entirely sure what echarts wants from me to make it all place nicelol

ryan953 commented on static/app/views/replays/detail/memoryChart.tsx:
```diff
@@ -50,9 +61,9 @@ function MemoryChart({memorySpans = []}: Props) {
       // we don't set a max because we let echarts figure it out for us
       axisLabel: {
-        formatter: value => formatBytesBase2(value),
+        formatter: value => getRoundedMemorySizeFromBytes(value),
```
I think we probably want to use the `interval` setting: https://echarts.apache.org/en/option.html#xAxis.axisLabel.interval

What we've got here looks like we're re-labelling the tick marks, which doesn't re-position them to match the new number.

Danondso replied:
I think I need some clarification here. If I use only the interval I don't get the formatting of the values to be rounded, I can go back to decimal formatting but that doesn't change the interval labels.

ryan953 replied:
I think we'll need both parts; `formatter` and `interval`.

Formatter is for sure needed to get the unit suffix for each label.
But interval is important too because it sets the distance between each label.

Before it was possible to have weird distances: 
<img width="88" alt="Screen Shot 2022-04-29 at 1 41 03 PM" src="https://user-images.githubusercontent.com/187460/166066343-56b847f4-4123-4174-b62a-ffc9090d5cff.png">
`19.1` to `28.6` is a distance of of `9.5`

For example sake lets say the total heap size is smaller, so the distance is (instead of 9.5) `4.5`. We'd have labels like:
- `22.5 MiB`
- `18 MiB`
- `13.5 MiB`
- `9 MiB`

If we just round those labels it'll be weird:
- `20 MiB`
- `20 MiB`
- `10 MiB`
- `10 MiB`

We also want to make sure that each point in the chart aligns with the axes labels. Currently they do.
This example shows 5.2 between 4.8 and 5.7, slightly closer to the one on the bottom.
<img width="351" alt="Screen Shot 2022-04-29 at 1 49 38 PM" src="https://user-images.githubusercontent.com/187460/166067384-b5ef1ece-e7cf-4d20-96c9-e884f4931c07.png">

Danondso replied:
I'll have to come back to this Monday, I think I almost have it but setting an interval doesn't change how the graph is rendering.

Danondso replied:
Okay I think I got it.. I set the minInterval to 1 and a busier looking chart seems to be correct. 
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/7014871/166244553-399da989-eea1-4fae-aff8-7d0a0ea19901.png">
Here's the same chart unrounded for reference. 
<img width="1863" alt="image" src="https://user-images.githubusercontent.com/7014871/166244608-98a5dfc8-121e-426f-9027-fdbef19c882d.png">

ryan953 replied:
So i'm still worried about this happening:

<img width="1412" alt="Screen Shot 2022-04-29 at 3 56 42 PM" src="https://user-images.githubusercontent.com/187460/166253638-e50ba019-8c59-4c21-97b0-059dbe4459af.png">

From: https://dev.getsentry.net:7999/organizations/sentry-emerging-tech/replays/sourcemapsio-replays:49798a94d812457ab4c2ce1f1d4d3532/#memory

But i've got the docs open to help sort out the specifics

ryan953 replied:
Chatted offline, learned about all the echarts options and @Danondso has some improvements coming.

Danondso committed:
fix?: adjusted interval so values that are rounded line up more correctly.

Danondso committed:
Merge branch 'master' into style/replays/34110/heap-memory-chart-followup

Danondso committed:
feat: added relative time (not shown correctly right now)
fix: corrected y-axis value for the memory chart so the ticks make sense
feat: added a position param to the bytes converter function so I can shave off the .0

ryan953 commented on static/app/views/replays/detail/memoryChart.tsx:
```diff
@@ -1,69 +1,108 @@
 
 const formatTimestamp = timestamp =>
-  getFormattedDate(timestamp * 1000, 'MMM D, YYYY HH:mm:ss');
+  getFormattedDate(timestamp * 1000, 'MMM D, YYYY hh:mm:ss A', {local: false}).concat(
+    ' UTC'
+  );
```
I think `z` inside the format string will work to replace the `.concat(' UTC')` call.

https://momentjs.com/docs/#/displaying/

```suggestion
  getFormattedDate(timestamp * 1000, 'MMM D, YYYY hh:mm:ss A z', {local: false});
```

ryan953 reviewed (approved):
This is all working for me 👯

Danondso committed:
Update static/app/views/replays/detail/memoryChart.tsx

Co-authored-by: Ryan Albrecht <ryan@ryanalbrecht.ca>

Danondso committed:
Merge branch 'master' into style/replays/34110/heap-memory-chart-followup

Danondso committed:
Merge branch 'master' into style/replays/34110/heap-memory-chart-followup

billyvg reviewed (approved):
Thanks for the follow-up!
